src/components/article/CommentPost.jsx

```diff
@@ -0,0 +1,68 @@
+import { useContext, useState } from "react";
+import { UsernameContext } from "../../contexts/Username";
+import { postArticleComment } from "../api";
+
+export default function CommentPost({article_id, handleViewComments}) {
+  const { username } = useContext(UsernameContext);
+  const [commentText, setCommentText] = useState("");
+  const [isPosting, setIsPosting] = useState(false);
+  const [isPosted, setIsPosted] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  function handleArticleComment(e) {
+    setCommentText(e.target.value);
+  }
+  function handleCommentSubmit(e){
+    setIsError(false)
+    e.preventDefault()
+    setIsPosting(true)
+    postArticleComment(article_id, username, commentText)
+    .then(()=>{
+        setIsPosting(false)
+        setIsPosted(true)
+        handleViewComments()
+        setTimeout(()=>{
+            setIsPosted(false)
+            setCommentText("")
+            e.target.disabled = false
+        },2000)
+    })
+    .catch((err)=>{
+        e.target.disabled = false
+        setIsError(true)
+    })
+    e.currentTarget.disabled=true
```

Comment:
I think that having line 34 before the post request would make this a bit more readable